x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsManager.java

```diff
@@ -108,6 +108,8 @@ private void reindexDataframeAndStartAnalysis(DataFrameAnalyticsTask task, DataF
             createIndexResponse -> {
                 ReindexRequest reindexRequest = new ReindexRequest();
                 reindexRequest.setSourceIndices(config.getSource());
+                // we default to match_all
+                reindexRequest.setSourceQuery(config.getParsedQuery());
```

Comment:
Actually, if we use the query while reindexing, we do not need the query at all at the data extractor. We might end up allowing working on an index without reindexing, in which case we'll want the query to be on the extractor instead, but while we're not doing that, the most efficient way is to indeed apply the query whilst reindexing.